scripts/pdf/generate_pdf.py

```diff
@@ -194,7 +194,7 @@ def write_content(content, output):
     for line in content.split(os.linesep):
         if "toctree" in line:
             break
-        line = f"#{line}" if (line.startswith("# ") or line.startswith("## ")) else line
+        line = f"#{line}" if line.startswith("#") else line
```

Comment:
This has more potential to mess with code blocks than the previous logic. e.g. it will mess with 

```c++
#include <stdio>
```

Reply:
I've changed to matching using regex, including trailing space.